src/BeamAdapter/component/engine/WireRestShape.inl

```diff
@@ -588,6 +618,14 @@ void WireRestShape<DataTypes>::initFromLoader()
         return;
     }
 
+    // this is... dirty but d_straightLength itself is not relevant for loader-created wire
+    // but the code uses it actively and expects it to not be zero.
+    if (d_straightLength.getValue() <= 0.0)
+    {
+        msg_warning() << "straightLength cannot be 0 (or negative...). Setting a minimum value.";
```

Comment:
In my mind, it should be possible to have only one "part" on your tool (because why not) so straightLength itself should be possible to be set to zero.... so here it is more of a "pis-allez" to circumvent the really bad design of length/straightlength